txnsync/bitmask.go

```diff
@@ -164,14 +164,15 @@ func (b *bitmask) iterate(entries int, maxIndex int, callback func(int, int) err
 		}
 	case 2:
 		sum := 0
-		for index := 0; index*2+2 < len(*b); index++ {
+		elementsCount := (len(*b) - 1) / 2
+		if elementsCount > maxIndex {
+			return errDataMissing
+		}
+		for index := 0; index < elementsCount; index++ {
 			sum += int((*b)[index*2+1])*256 + int((*b)[index*2+2])
-			if sum >= entries {
+			if sum > entries {
```

Comment:
I think this is not correct. 
Look at txnsync/encodedgroupsunmarshalers.go
`entries` is `int(stub.TotalTransactionsCount)`
and the `sum` passed to `callback` gets to `i`:
`signedTxns[i].Msig.Version`

Reply:
I agree with Shant. If sum = entries, we will be trying to access an index that is out of bounds.